### PR TITLE
fix/refactor: simplify db types and re-introduce `getOrCreateUser()`

### PR DESCRIPTION
**Pull request by iuioiua**
This change:
* Simplifies how database objects are get and set by including the `id` property in the respective object. This means we can return the object's value, including the `id`, ignoring the key.
* Re-introduces `getOrCreateUser()`, which ensures a user object always exists. This is particularly valuable for OAuth users.

Fixes:
* #133

### CODE REVIEW

File: utils/db.ts [hunk 1]
```diff
@@ -132,24 +137,24 @@ export async function setUserSubscription(
...
   const key = ["users", id];
-  const userRes = await kv.get<UserValue>(key);
+  const userRes = await kv.get<User>(key);
 
   if (userRes.value === null) throw new Error("User with ID does not exist");
 
   const res = await kv.atomic()
     .check(userRes)
-    .set(key, { ...userRes.value, isSubscribed } as UserValue)
+    .set(key, { ...userRes.value, isSubscribed } as User)
```
Comment (kt3k) [inline utils/db.ts:146]: This seems only updating the user at `users, id` key and the other user data at `users_by_stripe_customer` is not update. I don't think saving the same entity (user) in 2 different places are good idea (That will be bug prone, and the broken data will be hard to fix.)

File: utils/db.ts [hunk 2]
```diff
@@ -1,127 +1,132 @@
...
 
-export async function getUserIdByStripeCustomerId(stripeCustomerId: string) {
-  const res = await kv.get<UserValue["id"]>([
-    "user_ids_by_stripe_customer",
+export async function getUserById(id: string) {
+  const res = await kv.get<User>(["users", id]);
+  return res.value;
+}
+
+export async function getUserByStripeCustomerId(stripeCustomerId: string) {
```
Comment (kt3k) [inline utils/db.ts:127]: Let's make this function call `kv.get` twice, such as:
- `kv.get("user_id_by_stripe_customer_id", stripeCustomerId)` -> userId
- `kv.get("users", userId)` -> User

Comment (iuioiua) [reply utils/db.ts:127]: Done.

**Review by kt3k — APPROVED** [review]
LGTM

**Comment by kt3k** [comment]
@iuioiua 
Not directly the issue of this change, but `comments` and `items` are also saved in 2 keys and I think those will be hard to manage in the future (especially when we allow users to update them).

I think we should store only item ids in `item_ids_by_user_id, <user_id>, item_id` key and get them like:
- `kv.list({ prefix: ["item_ids_by_user_id", userId] })` -> list of itemIds
- `kv.getMany([["items", itemId0], ["items", itemId1], ...])` -> list of Items

**Comment by iuioiua** [comment]
Yes, this is the challenge of working with relations within a key-value store. A few clever utilities should solve this. I'll see what I can come up with.

**Pull request merged**